src/main/scala/org/scalacheck/MasterRunner.scala

```diff
@@ -0,0 +1,67 @@
+package org.scalacheck
+
+
+import sbt.testing._
+import java.util.concurrent.atomic.AtomicInteger
+
+class  MasterRunner(
+                    args: Array[String],
+                    remoteArgs: Array[String],
+                    loader: ClassLoader)
+  extends BaseRunner(args, remoteArgs, loader) {
+
+
+  /** Number of tasks registered in the whole system */
+  private[this] val registeredCount = new AtomicInteger(0)
+
+  /** Number of running slaves in the whole system */
+  private[this] val slaveCount = new AtomicInteger(0)
+
+  def tasks(taskDefs: Array[TaskDef]): Array[Task] = {
+    registeredCount.addAndGet(taskDefs.length)
+    taskDefs.map(newTask)
+  }
+
+  def done(): String = {
+    val slaves = slaveCount.get
+    val registered = registeredCount.get
+    val done = results.doneCount.get
+    val test = results.testCount.get
+    val success = results.successCount.get
+    val failed = results.failureCount.get
+    val errors = results.errorCount.get
+
+    if (slaves > 0) {
+      throw new IllegalStateException(
+        s"There are still $slaves slaves running")
+    }
```

Comment:
Actually, maybe you don't even need a `MasterRunner` and a `SlaveRunner`. [MiniTest does it all with one single `Runner` class](https://github.com/monifu/minitest/tree/master/shared/main/scala/minitest/runner)